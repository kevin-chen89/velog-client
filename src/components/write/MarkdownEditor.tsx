import * as React from 'react';
import styled from 'styled-components';
import TitleTextarea from './TitleTextarea';

export interface MarkdownEditorProps {}

const MarkdownEditorBlock = styled.div``;
export default class MarkdownEditor extends React.Component<
  MarkdownEditorProps,
  any
> {
  public render() {
    return (
      <MarkdownEditorBlock>
        <TitleTextarea />
      </MarkdownEditorBlock>
    );
  }
}
