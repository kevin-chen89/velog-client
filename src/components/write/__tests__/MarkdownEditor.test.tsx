import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';
import MarkdownEditor, { MarkdownEditorProps } from '../MarkdownEditor';

describe('MarkdownEditor', () => {
  const setup = (props: Partial<MarkdownEditorProps> = {}) => {
    const initialProps: MarkdownEditorProps = {
      onChangeMarkdown: () => {},
      onChangeTitle: () => {},
      title: '',
    };
    const utils = render(<MarkdownEditor {...initialProps} {...props} />);
    const titleTextarea = utils.getByPlaceholderText(
      '제목을 입력하세요',
    ) as HTMLTextAreaElement;
    return {
      titleTextarea,
      ...utils,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('has codemirror editor', () => {
    const utils = setup();
    utils.getByText('당신의 이야기를 적어보세요...');
  });
  it('changes title properly', () => {
    const onChangeTitle = jest.fn();
    const utils = setup({
      onChangeTitle,
    });
    fireEvent.change(utils.titleTextarea, {
      target: {
        value: 'changed',
      },
    });
    expect(onChangeTitle).toBeCalledWith('changed');
    // expect(utils.titleTextarea.value).toBe('changed!');
  });
  it('uses title prop', () => {
    const utils = setup({
      title: 'hello world',
    });
    expect(utils.titleTextarea.value).toBe('hello world');
  });
});
