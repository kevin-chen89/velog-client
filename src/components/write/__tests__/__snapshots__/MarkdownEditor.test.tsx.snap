// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownEditor matches snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH erJALV"
  >
    <div
      class="wrapper"
    >
      <textarea
        class="sc-bdVaJa cuPLSE"
        placeholder="제목을 입력하세요"
        style="height: 0px;"
      />
    </div>
  </div>
</div>
`;
